Orange/data/io.py

```diff
@@ -337,248 +349,169 @@ def open(filename, *args, **kwargs):
+                        raise ValueError("Non-continuous data read into a continuous column "
+                                         "(column {}: {}).".format(col_idx + 1, name))
+
+                # use an existing variable if available, otherwise get a brand new one
+                # with a brand new name
+                new_name = Domain.infer_name(col_type, col_role, name,
+                                             role_vars['x'], role_vars['y'], role_vars['meta'])
+                if name == new_name:
+                    var = col_type.make(new_name, **col_type_kwargs)
                 else:
-                    # Never use existing for un-named variables
-                    var = coltype(next(NAMEGEN), **coltype_kwargs)
+                    var = col_type(new_name, **col_type_kwargs)
+                # also store any attributes in the third row (beside the role declaration)
                 var.attributes.update(flag.attributes)
-                domain_vars.append(var)
-
-                # Reorder discrete values to match existing variable
-                if var.is_discrete and not var.ordered:
-                    new_order, old_order = var.values, coltype_kwargs.get('values', var.values)
-                    if new_order != old_order:
-                        offset = len(new_order)
-                        column = values if data.ndim > 1 else data
-                        column += offset
-                        for i, val in enumerate(var.values):
-                            try: oldval = old_order.index(val)
-                            except ValueError: continue
-                            bn.replace(column, offset + oldval, new_order.index(val))
-
-            if coltype is TimeVariable:
-                # Re-parse the values because only now after coltype.make call
-                # above, variable var is the correct one
-                values = [var.parse(i) for i in orig_values]
-
-            # Write back the changed data. This is needeed to pass the
-            # correct, converted values into Table.from_numpy below
-            try: data[:, col] = values
-            except IndexError: pass
-
-        domain = Domain(attrs, clses, metas)
-
-        if not data.size:
-            return Table.from_domain(domain, 0)
-
-        table = Table.from_numpy(domain,
-                                 data[:, Xcols].astype(float, order='C'),
-                                 data[:, Ycols].astype(float, order='C'),
-                                 data[:, Mcols].astype(object, order='C'),
-                                 data[:, Wcols].astype(float, order='C'))
-        return table
+
+                role_vars[col_role].append(var)
+                # strip whitespace from string/string-like columns (np.object_ in pandas)
+                if np.issubdtype(contents[col_idx], np.object_):
+                    result[var.name] = contents[col_idx].str.strip()
```

Comment:
It can't be, pandas doesn't eval. 
```python
pd.read_csv(StringIO("[1,2,3] ;1; 2; 3"), header=None, sep=';')[0].str.strip()
0    [1,2,3]
Name: 0, dtype: object
```